src/realm/group.hpp

```diff
@@ -1002,7 +1002,7 @@ inline ref_type Group::get_history_ref(const Array& top) noexcept
 inline int Group::get_history_schema_version(const Array& top) noexcept
 {
     if (top.is_attached()) {
-        if (top.size() >= 8) {
+        if (top.size() >= 9) {
```

Comment:
Ok, I have taken another look at this, and the change (8->9) no longer makes sense to me. The logical meaning of the condition is "does this file have a changeset history or not". As can be seen in  https://github.com/realm/realm-core/blob/c1dcd7c6376c868563475535a975b6aebd7e43a6/src/realm/group.hpp#L545, both the 8th and the 9th entry were introduced together in file format version 4, and they will always either both be there, or both not be there. This means that `top.size() >= 8` and `top.size() >= 9` effectively check the same thing, and that the change above makes no difference. Still I'd prefer the version with 8, as that makes most sense to me. @tgoyne Can you cast some light on why you changed it? For now, I'll add a commit to this PR that introduces a named boolean to make the purpose of the condition more clear.